modules/save_contigs_per_kingdom.nf

```diff
@@ -1,17 +1,16 @@
 process SAVE_CONTIGS_PER_KINGDOM {
-  label 'light'
+  label 'process_single'
 
   input:
-  val contigsPath
-  each result
+  path contigsPath
+  each ids
```

Comment:
Just a minor comment here. In my pipelines when I don't want to run a process per file (because it will spawn lots of jobs in the cluster), I put the files in a folder and feed that folder path to my process instead. Then I call a python script from the module that parses/executes the files one by one, putting their results in one main folder, again which will be the output of my module. I think that would be equal functionality to how you use 'each'? But not sure since I've never used it. If it works as expected here, leave it as it is of course. Just make sure you get the resutl you would expect ;)

Reply:
I didn't thought about this possibiliy thanks, it can be really useful !
The `each ids` allows me here to run the module for each path in the input path channel individually. As each path is a file representing a predicted kingdom and containing the IDs of associated contigs, it can only be launched maximum 4 times (eukaryotes, prokaryotes, unknwon, requiringCATvalidation), so it's not a lot, but your idea could definitively be useful in larger cases ! :)